analyzers/tests/SonarAnalyzer.UnitTest/Extensions/KnownReferenceTests.cs

```diff
@@ -0,0 +1,295 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Moq;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Extensions;
+using static SonarAnalyzer.Helpers.KnownReference;
+using static SonarAnalyzer.Helpers.KnownReference.Predicates;
+
+namespace SonarAnalyzer.UnitTest;
+
+[TestClass]
+public class KnownReferenceTests
+{
+    [TestMethod]
+    public void KnownReference_ThrowsWhenPredicateNull_1()
+    {
+        var sut = () => new KnownReference(default(Func<AssemblyIdentity, bool>));
+        sut.Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("predicate");
+    }
+
+    [TestMethod]
+    public void KnownReference_ThrowsWhenPredicateNull_2()
+    {
+        var sut = () => new KnownReference(default(Func<IEnumerable<AssemblyIdentity>, bool>));
+        sut.Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("predicate");
+    }
+
+    [TestMethod]
+    public void KnownReference_ThrowsWhenPredicateNull_Params()
+    {
+        var sut = () => new KnownReference(_ => true, null, _ => true);
+        sut.Should().Throw<ArgumentNullException>().Which.ParamName.Should().Be("predicate");
+    }
+
+    [DataTestMethod]
+    [DataRow("Test", true)]
+    [DataRow("test", true)]
+    [DataRow("TEST", true)]
+    [DataRow("MyTest", false)]
+    [DataRow("TestMy", false)]
+    [DataRow("MyTestMy", false)]
+    [DataRow("MyTESTMy", false)]
+    [DataRow("Without", false)]
+    public void NameIs_Test(string name, bool expected)
+    {
+        var sut = new KnownReference(NameIs("Test"));
+        var compilation = new Mock<Compilation>("compilationName", ImmutableArray<MetadataReference>.Empty, new Dictionary<string, string>(), false, null, null);
+        var identity = new AssemblyIdentity(name);
+        compilation.SetupGet(x => x.ReferencedAssemblyNames).Returns(new[] { identity });
+        sut.IsReferencedBy(compilation.Object).Should().Be(expected);
```

Comment:
Let's extract the repetitive scaffodling into a static method
```suggestion
        sut.IsReferencedBy(CreateCompilation(name)).Should().Be(expected);
```

Reply:
Version can be an optional parameter